### ISSUE TITLE
s3 select api response payload can't be parsed

### ISSUE THREAD

**Issue by pritamsarkar007** [terrycain/aioboto3#178]
* Async AWS SDK for Python version: aioboto3(6.4.1), aiobotocore(0.10.3)
* Python version: 3.7
* Operating System: MACOS 10.14.5
asyncio (3.4.3)
### Description
I am new to python, so pardon me if I am doing something horribly wrong. 
I am trying to call s3 select api asynchronously and getting a successful response but not able to iterate over the payload. The EventStream returned by the call may not be compatible for asynchronous calls.

### What I Did
I tried two different ways and none of those worked. Note that the prints in the code section below is working fine
option1: 
command:
```python <my_file_name>```
code
```
session = aioboto3.Session(profile_name='<profile_name>', region_name='us-east-1')    
async with session.client("s3") as client:
        obj = await client.select_object_content(**kwargs)
        print(obj['Payload'])
        print(sys.getsizeof(obj['Payload']))
        print(await obj['Payload']._raw_stream.read())


        async with obj['Payload'] as event_stream:
            for event in event_stream:
                if 'Records' in event:
                    data = event['Records']['Payload'].decode('utf-8')
                    print(data)
```
stack trace
```
Traceback (most recent call last):
  File "s3_util.py", line 164, in <module>
    asyncio.run(main())
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "s3_util.py", line 159, in main
    'where \'aws_route53_record\' in s.resources[*].type')
  File "s3_util.py", line 53, in s3_select_from_object
    async with obj['Payload'] as event_stream:
AttributeError: __aexit__

```

option 2:
command: ```python <my_file_name>```
code:
```
session = aioboto3.Session(profile_name='<profile_name>', region_name='us-east-1')     
async with session.client("s3") as client:
        obj = await client.select_object_content(**kwargs)
        print(obj['Payload'])
        print(sys.getsizeof(obj['Payload']))
        print(await obj['Payload']._raw_stream.read())

        for event in obj['Payload']:
            if 'Records' in event:
                data = event['Records']['Payload'].decode('utf-8')
                print(data)
        async with obj['Payload'] as event_stream:
            for event in event_stream:
                if 'Records' in event:
                    data = event['Records']['Payload'].decode('utf-8')
                    print(data)
```
stack trace:
```
Traceback (most recent call last):
  File "s3_util.py", line 167, in <module>
    asyncio.run(main())
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "s3_util.py", line 162, in main
    'where \'aws_route53_record\' in s.resources[*].type')
  File "s3_util.py", line 52, in s3_select_from_object
    for event in obj['Payload']:
  File "/Users/lrd377/PycharmProjects/emr-cluster-init/venv/lib/python3.7/site-packages/botocore/eventstream.py", line 570, in __iter__
    for event in self._event_generator:
  File "/Users/lrd377/PycharmProjects/emr-cluster-init/venv/lib/python3.7/site-packages/botocore/eventstream.py", line 577, in _create_raw_event_generator
    for chunk in self._raw_stream.stream():
AttributeError: 'StreamReader' object has no attribute 'stream'

```

**Comment by terrycain** [comment]
So going down the rabbit hole, I'll look into updating aiobotocore to deal with this new stuff https://github.com/boto/botocore/blob/1ae092ed95b39c3336ba48c5cbd71b7c37924d05/botocore/parsers.py as those parsers will need to be asyncified.

**Comment by terrycain** [comment]
Ok I have a fix, is not out yet but once its upstream, that should solve this and any future aws api that uses event streams. You'll then be able to use 

```python
session = aioboto3.Session(profile_name='<profile_name>', region_name='us-east-1')    
async with session.client("s3") as client:
        obj = await client.select_object_content(**kwargs)
        async for event in obj['Payload']:
            if 'Records' in event:
                print(event['Records'])
```

**Comment by terrycain** [comment]
Should be fixed by v6.5.0 can you test it out? If it doesn't work can you reopen this ticket with the issue.
